modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalBasicWithFieldName.tsx

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import ClayForm from '@clayui/form';
+import ClayModal from '@clayui/modal';
+import React, {useState} from 'react';
+
+import useForm from '../hooks/useForm';
+import Input from './form/Input';
+
+interface IProps extends React.HTMLAttributes<HTMLElement> {
+	apiURL: string;
+	inputId: string;
+	label: string;
+	observer: any;
+	onClose: () => void;
+}
+
+type TInitialValues = {
+	name: {
+		[key: string]: string;
+	};
+};
+
+const defaultLanguageId = Liferay.ThemeDisplay.getDefaultLanguageId();
+
+const labels = {
+	newLayout: Liferay.Language.get('new-layout'),
+	newView: Liferay.Language.get('new-view'),
+};
+
+const ModalBasicWithFieldName: React.FC<IProps> = ({
+	apiURL,
+	inputId,
+	label,
+	observer,
+	onClose,
+}) => {
+	const initialValues: TInitialValues = {
+		name: {
+			[defaultLanguageId]: '',
+		},
+	};
+	const [error, setError] = useState<string>('');
+
+	const onSubmit = async ({name}: TInitialValues) => {
+		const response = await Liferay.Util.fetch(apiURL, {
+			body: JSON.stringify({
+				name: {
+					[defaultLanguageId]: name,
+				},
+			}),
+			headers: new Headers({
+				'Accept': 'application/json',
+				'Content-Type': 'application/json',
+			}),
+			method: 'POST',
+		});
+
+		if (response.status === 401) {
+			window.location.reload();
+		}
+		else if (response.ok) {
+			onClose();
+
+			window.location.reload();
+		}
+		else {
+			const {
+				title = Liferay.Language.get('an-error-occurred'),
+			} = await response.json();
+
+			setError(title);
+		}
+	};
+
+	const validate = ({name}: TInitialValues) => {
+		const errors: any = {};
```

Comment:
The more strict the better!

avoid use `any` type:
```suggestion
		const errors: {name?: string} = {};
```